File: packages/walletconnect_pay/android/src/main/kotlin/com/walletconnect/walletconnect_pay/WalletconnectPayPlugin.kt
package com.walletconnect.walletconnect_pay

import io.flutter.embedding.engine.plugins.FlutterPlugin
import io.flutter.plugin.common.MethodCall
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.MethodChannel.MethodCallHandler
import io.flutter.plugin.common.MethodChannel.Result

/** WalletconnectPayPlugin */
class WalletconnectPayPlugin : FlutterPlugin, MethodCallHandler {
    private lateinit var channel: MethodChannel

    override fun onAttachedToEngine(flutterPluginBinding: FlutterPlugin.FlutterPluginBinding) {
        channel = MethodChannel(flutterPluginBinding.binaryMessenger, "walletconnect_pay")
        channel.setMethodCallHandler(this)
    }

    override fun onMethodCall(call: MethodCall, result: Result) {
        when (call.method) {
            "initialize" -> WalletConnectPayClient.initialize(call.arguments, result)
            "confirmPayment" -> WalletConnectPayClient.confirmPayment(call.arguments, result)
            "getPaymentOptions" -> WalletConnectPayClient.getPaymentOptions(call.arguments, result)
            "getRequiredPaymentActions" -> WalletConnectPayClient.getRequiredPaymentActions(call.arguments, result)
            else -> result.notImplemented()
        }
    }

    override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {
        channel.setMethodCallHandler(null)
    }
}

File: packages/walletconnect_pay/ios/Classes/WalletconnectPayPlugin.swift
import Flutter
import UIKit

public class WalletconnectPayPlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "walletconnect_pay", binaryMessenger: registrar.messenger())
    let instance = WalletconnectPayPlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "initialize":
      WalletConnectPayClient.initialize(params: call.arguments, result: result)
    case "confirmPayment":
      WalletConnectPayClient.confirmPayment(params: call.arguments, result: result)
    case "getPaymentOptions":
      WalletConnectPayClient.getPaymentOptions(params: call.arguments, result: result)
    case "getRequiredPaymentActions":
      WalletConnectPayClient.getRequiredPaymentActions(params: call.arguments, result: result)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
